diff --git a/plugins/dde-sys-monitor-plugin/CMakeLists.txt b/plugins/dde-sys-monitor-plugin/CMakeLists.txt
index 61d7fbc..25fef2e 100755
--- a/plugins/dde-sys-monitor-plugin/CMakeLists.txt
+++ b/plugins/dde-sys-monitor-plugin/CMakeLists.txt
@@ -38,7 +38,6 @@ file(GLOB_RECURSE SRCS "*.h" "*.cpp" "*.ui")
 # <库名>_INCLUDE_DIRS   有哪些头文件目录（Qt5Widgets_INCLUDE_DIRS）
 # <库名>_LIBRARIES      有哪些库文件（Qt5Widgets_LIBRARIES）
 find_package(Qt6 REQUIRED COMPONENTS Widgets)
-pkg_check_modules(dtk2widget REQUIRED dtk2widget)
 
 # find_package 命令还可以用来加载 cmake 的功能模块
 # 并不是所有的库都直接支持 cmake 查找的，但大部分都支持了 pkg-config 这个标准，
@@ -49,6 +48,7 @@ pkg_check_modules(dtk2widget REQUIRED dtk2widget)
 # PKG_CONFIG_EXECUTABLE       pkg-config 可执行文件的路径
 # PKG_CONFIG_VERSION_STRING   pkg-config 的版本信息
 find_package(PkgConfig REQUIRED)
+pkg_check_modules(dtk2widget REQUIRED dtk2widget)
 
 # 加载 FindPkgConfig 模块后就可以使用 pkg_check_modules 命令加载需要的库
 # pkg_check_modules 命令是由 FindPkgConfig 模块提供的，因此要使用这个命令必须先加载 FindPkgConfig 模块。
