diff --git a/platformthemeplugin/qdeepintheme.cpp b/platformthemeplugin/qdeepintheme.cpp
index 4440e61..f1611df 100644
--- a/platformthemeplugin/qdeepintheme.cpp
+++ b/platformthemeplugin/qdeepintheme.cpp
@@ -158,7 +158,11 @@ static void updateWindowGeometry(QWindow *w)
         return;
 
     if (w->property(DNOT_UPDATE_WINDOW_GEOMETRY).toBool()) {
-        QWindowSystemInterfacePrivate::GeometryChangeEvent gce(w, QHighDpi::fromNativePixels(w->handle()->geometry(), w)
+        QWindowSystemInterfacePrivate::GeometryChangeEvent gce(w
+                                                   #if QT_VERSION >= QT_VERSION_CHECK(6, 9, 2)
+                                                               , QHighDpi::fromNativeWindowGeometry(w->handle()->geometry(), w)
+                                                   #endif
+                                                               , QHighDpi::fromNativePixels(w->handle()->geometry(), w)
                                                    #if QT_VERSION < QT_VERSION_CHECK(5, 10, 0)
                                                                , QRect()
                                                    #endif
